native/src/modules/endpoint/sagas/watchFetchCategory.js

```diff
@@ -42,8 +43,11 @@ export function * fetchCategory (dataContainer: DataContainer, action: FetchCate
         call(dataContainer.getResourceCache, city, language)
       ])
 
-      const push: PushCategoryActionType = {
-        type: 'PUSH_CATEGORY',
+      const lastUpdate: Moment | null = yield call(dataContainer.getLastUpdate, city, language)
+
+      // $FlowFixMe Flow can't evaluate the type as it is dynamic
```

Comment:
That actually should work. I think it is possible to refactor this that it works. You can create a function with return type `PushCategoryActionType | RefreshCategoryActionType`. Inside that function you have a if which returns one action or the other one.

Reply:
Yeah I had that idea, too. Didn't like it though, as it unnecessarily makes the code more verbose and I also don't like having to adjust my code just because of flow's weaknesses :/
I also mentioned this in some comment above.

Reply:
Irrelevant now as I unified the actions.